packages/node-http-handler/src/node-http2-handler.spec.ts

```diff
@@ -28,288 +29,420 @@ describe("NodeHttp2Handler", () => {
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        const authority = `${protocol}//${hostname}:${port}`;
+        expect(connectSpy).toHaveBeenCalledTimes(1);
+        expect(connectSpy).toHaveBeenCalledWith(authority);
+      });
+
+      it("is many if multiple requests are made on same URL", async () => {
+        const connectSpy = jest.spyOn(http2, "connect");
+
+        // Make two requests.
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        const authority = `${protocol}//${hostname}:${port}`;
+        expect(connectSpy).toHaveBeenCalledTimes(2);
+        expect(connectSpy).toHaveBeenNthCalledWith(1, authority);
+        expect(connectSpy).toHaveBeenNthCalledWith(2, authority);
+      });
+
+      it("is many if requests are made on different URLs", async () => {
+        const connectSpy = jest.spyOn(http2, "connect");
+
+        // Make first request on default URL.
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        const port2 = port + 1;
+        const mockH2Server2 = createMockHttp2Server().listen(port2);
+        mockH2Server2.on("request", createResponseFunction(mockResponse));
+
+        // Make second request on URL with port2.
+        await nodeH2Handler.handle(new HttpRequest({ ...getMockReqOptions(), port: port2 }), {});
+
+        const authorityPrefix = `${protocol}//${hostname}`;
+        expect(connectSpy).toHaveBeenCalledTimes(2);
+        expect(connectSpy).toHaveBeenNthCalledWith(1, `${authorityPrefix}:${port}`);
+        expect(connectSpy).toHaveBeenNthCalledWith(2, `${authorityPrefix}:${port2}`);
+        mockH2Server2.close();
+      });
     });
 
-    /* Commenting out as the test is flaky https://github.com/aws/aws-sdk-js-v3/issues/487
-    it("will close request on session when aborted", async () => {
-      await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
-
-      // @ts-ignore: access private property
-      const session: ClientHttp2Session = nodeH2Handler.connectionPool.get(
-        `${protocol}//${hostname}:${port}`
-      );
-      const requestSpy = jest.spyOn(session, "request");
-
-      const abortController = new AbortController();
-      // Delay response so that onabort is called earlier
-      setTimeout(() => {
-        abortController.abort();
-      }, 0);
-      mockH2Server.on(
-        "request",
-        async () =>
-          new Promise(resolve => {
-            setTimeout(() => {
-              resolve(createResponseFunction(mockResponse));
-            }, 1000);
-          })
-      );
-
-      await expect(
-        nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {
-          abortSignal: abortController.signal
-        })
-      ).rejects.toHaveProperty("name", "AbortError");
-      expect(requestSpy.mock.calls.length).toBe(1);
+    describe("destroy", () => {
```

Comment:
duplicated to line 216?

Reply:
No. This section is for `disableConcurrentStreams: true`
The section on line 216 is for `disableConcurrentStreams: false`